### PR TITLE
tbi(environment-check): search node path for installed gens

### PR DESCRIPTION
**Pull request by IainSAP**
TBI: #866 

- Removes hardcoded BAS generator search paths
- Adds NODE_PATH search paths
- Update/adds tests

### CODE REVIEW

**Review by longieirl — DISMISSED** [review]

**Review by cianmSAP — APPROVED** [review]
Tested on BAS ✅

File: packages/environment-check/src/checks/get-installed.ts [hunk 1]
```diff
@@ -168,10 +166,13 @@ export async function getFioriGenVersion(): Promise<string> {
         const fioriGenPath = await getFioriGenGlobalPath();
         genSearchPaths.push(fioriGenPath);
 
-        if (isAppStudio()) {
-            const fioriGenPathBAS = join(globalNodeModulesBAS, NpmModules.FioriGenerator);
-            const fioriGenPathPnpmBAS = join(globalNodeModulePnpmBAS, NpmModules.FioriGenerator);
-            genSearchPaths.push(fioriGenPathBAS, fioriGenPathPnpmBAS);
+        // BAS recommend using NODE_PATH as the paths therein are used to locate all generators regardless of installation method
+        if (isAppStudio() && process.env.NODE_PATH) {
```
Comment (ullasholla) [inline packages/environment-check/src/checks/get-installed.ts:170]: Is `process.env.NODE_PATH` guaranteed to be set on BAS? If so, why check it? If not, what happens when it's not set?

Is there doco from BAS about this somewhere?

Comment (IainSAP) [reply packages/environment-check/src/checks/get-installed.ts:170]: I have added tests to cover it not being set. It should be set, but better to be safe. The only doc is emails. It was their suggestion.

**Review by devinea — APPROVED** [review]
updates to code and tests lg.
Thanks @IainSAP

**Review by Klaus-Keller — APPROVED** [review]
- code changes look good
- test coverage is great
- did just visual review, did not build and test

Approved from my side.

**Review by cianmSAP — APPROVED** [review]

**Pull request merged**